Dev query2

tomidahab opened the pull request.

tomidahab committed:
added new worker for query2

tomidahab committed:
fix response to boundary

ivanovic99 reviewed (changes requested):
Hay un par de cositas.
Primero que nada, la query dos debria estar dividida en, a priori, dos workers.
El primero deberia iterar las movies y te quedas con el/los paises que EN ESE batch no colaborarron en alguna movie con algun otro pais (es decir que el production_countries de la movie tiene un len == 1) y manda ese nuevo batch filtrado al siguiente worker que le corresponde (hay que hacer algun especie de broker aca que pueda saber que worker proximo tiene que recibir esta respuesta, es decir, el worker que esta procesando la consulta del client X, no mandarlo a un worker que no esta procesando al cliente X).

Este proximo worker tiene un estado (statefull) el cual en este mantiene a todos los paises que no se repiten hasta el momento (junto con el budget y, si el nuevo batch contiene un pais el cual este worker ya continene en su estado, elimina de su estado a este pais repetido. Y para el resto, recalcula el budget. La estructura de datos que mantiene este worker (y que le manda el worker anterior) puede ser algo asi:
{
P1: [repetido, budget],
....
PN: [repetido, budget]
}

ivanovic99 committed:
Query 1 Done (#3)

* Add: .gitignore with python template + README.md

Co-authored-by: Regazzoli Ignacio <iregazzoli@users.noreply.github.com>

* Fix: gitignore csvs

* Add: .gitignore python template

Co-authored-by: Regazzoli Ignacio <iregazzoli@users.noreply.github.com>

* Update: Client sends in batches the csvs

Co-authored-by: Regazzoli Ignacio <iregazzoli@users.noreply.github.com>

* Add: basic setup.
Client connects to Boundary and sends csv.
Boundary sends csv (batches) to worker via rabbitmq

* Delete: pycache

* Update: batch reads only complete rows. If that last rows exceeds the batch_size, returns the read rows and resumes the batch reads from that last row

* Update: Skipping first line of csv (header skipped)

* Update: recursive env files ignored

Co-authored-by: Regazzoli Ignacio <iregazzoli@users.noreply.github.com>

* Add: Column projection and dummy serializer

Co-authored-by: Regazzoli Ignacio <iregazzoli@users.noreply.github.com>

* Add: Year filter logic

Co-authored-by: Regazzoli Ignacio <iregazzoli@users.noreply.github.com>

* Update: deleting RELEASE_DATE field after filtration

* Update: CREW changed for CREDITS

* Add: filter by country worker WIP

* Fix: filters logic for treating year_filter with each respective country_filter

Co-authored-by: Regazzoli Ignacio <iregazzoli@users.noreply.github.com>

* Add: Response manager task and writing response into file

Co-authored-by: Regazzoli Ignacio <iregazzoli@users.noreply.github.com>

* Fis: filter for query 1 (but does not match with catedra's kaggle though)

* Fix: .env (example) fix for crew --> credits

* Update: writning response into output file for testing

* Update: Separated each client into two threads, Receiver (infinite recv loop unless shutdown) and Sender (only sends the csvs) and writing the response un a file

* Update: catching _send_all error

* Add: client uuid metadata and sending formatted response to client

---------

Co-authored-by: Regazzoli Ignacio <iregazzoli@users.noreply.github.com>

tomidahab committed:
added new worker to divide responsabilities

tomidahab committed:
Merge branch 'main' into DEV-query2

tomidahab committed:
fix query2 problems

tomidahab committed:
added client id to requests in query2
